ibc-apps/ics20-transfer/types/src/coin.rs

```diff
@@ -41,7 +41,7 @@ pub struct Coin<D> {
 
 impl<D: FromStr> Coin<D>
 where
-    D::Err: Into<DecodingError>,
+    D::Err: Display,
```

Comment:
Just curious what the rationale was for changing the constraint here from `Into<DecodingError>` to `Display`.

Reply:
Check out https://github.com/cosmos/ibc-rs/pull/1335#discussion_r1752631075. The change was made to move away from using `From<Infallible> for DecodingError`. Also, I noticed that requiring `Into<DecodingError>` was a bit too restrictive for anyone wanting to define a new type of `Coin<D>`. Using `Display` is a more flexible and good enough requirement for ibc-rs internals.